### PR TITLE
char-rnn on gpu

### PR DESCRIPTION
**Pull request by gustafsson**
Takes 5 minutes on a GPU, vs, 60 minutes on four CPUs. Both train the loss to about 80-85 and produce similar sample output.

### CODE REVIEW

**Comment by gustafsson** [comment]
`crossentropy` uses `log` which isn't supported by CuArrays (https://github.com/JuliaGPU/CuArrays.jl/issues/71) so I changed it to use `logitcrossentropy` instead.

**Comment by gustafsson** [comment]
Changed back from `logitcrossentropy` to `crossentropy` since https://github.com/FluxML/Flux.jl/commit/8f73dc6e148eedd11463571a0a8215fd87e7e05b. Got the same performance results and similar output.

**Comment by DilumAluthge** [comment]
Any update on merging this PR? @gustafsson @MikeInnes

**Comment by gustafsson** [comment]
awaiting review, don't know if there's much else I can do

**Comment by MikeInnes** [comment]
Sorry for the wait, but thanks for the great patch!
